solana/solana-ibc/programs/solana-ibc/src/transfer/impls.rs

```diff
@@ -1,26 +1,131 @@
+use std::str::FromStr;
+
+use anchor_lang::prelude::{AccountInfo, CpiContext, Pubkey};
 use anchor_lang::solana_program::msg;
+use anchor_lang::AccountDeserialize;
+use anchor_spl::token::{spl_token, Burn, MintTo, TokenAccount, Transfer};
 use ibc::applications::transfer::context::{
     TokenTransferExecutionContext, TokenTransferValidationContext,
 };
 use ibc::applications::transfer::error::TokenTransferError;
-use ibc::applications::transfer::PrefixedCoin;
+use ibc::applications::transfer::{Amount, PrefixedCoin};
 use ibc::core::ics24_host::identifier::{ChannelId, PortId};
-use ibc::Signer;
+use primitive_types::U256;
+use strum::Display;
+use uint::FromDecStrErr;
 
-// use crate::module_holder::IbcStorage<'_,'_>;
+use crate::storage::ids::PortChannelPK;
 use crate::storage::IbcStorage;
+use crate::MINT_ESCROW_SEED;
+
+/// Structure to identify if the account is escrow or not. If it is escrow account, we derive the escrow account using port-id, channel-id and denom.
+#[derive(
+    Clone, Display, PartialEq, Eq, derive_more::From, derive_more::TryInto,
+)]
+pub enum AccountId {
+    Signer(Pubkey),
+    Escrow(PortChannelPK),
+}
+
+impl TryFrom<ibc::Signer> for AccountId {
+    type Error = <Pubkey as FromStr>::Err;
+
+    fn try_from(value: ibc::Signer) -> Result<Self, Self::Error> {
+        Ok(Pubkey::try_from(value.as_ref()).map(Self::Signer).unwrap())
+    }
```

Comment:
This is confusing.  For one why `unwrap` rather than returning the
error?  Furthermore, you’re using Pubkey::try_from but taking error
from FromStr trait.  Perhaps:

```suggestion
    type Error = <Pubkey as FromStr>::Err;

    fn try_from(value: ibc::Signer) -> Result<Self, Self::Error> {
        Pubkey::from_str(value.as_ref()).map(Self::Signer)
    }
```

Reply:
Yes i think the above makes sense.